src/setfit/modeling.py

```diff
@@ -243,13 +244,29 @@ def fit(
             self.model_head.fit(embeddings, y_train)
 
     def _prepare_dataloader(
-        self, x_train: List[str], y_train: List[int], batch_size: int, shuffle: bool = True
+        self,
+        x_train: List[str],
+        y_train: List[int],
+        batch_size: int,
+        max_length: Optional[int] = None,
+        shuffle: bool = True,
     ) -> DataLoader:
+        max_acceptable_length = self.model_body.get_max_seq_length()
+        max_length = max_length or max_acceptable_length
```

Comment:
I think it would be nice to be explicit here:

```suggestion
        if max_length is None:
            max_length = max_acceptable_length
```

Perhaps we can also log a warning that we're using `max_acceptable_length` in this case?

Reply:
good point!